### PR TITLE
Unslash the $_GET, $_POST, $COOKIE and $_SERVER variables 

### LINKED ISSUE #220: Quotes shouldn't be escaped
**Problem description**
When adding a translation that contains quotes, or when filtering on something containing a quote, the quote is automagically escaped. This can lead to problems in the translated languages.

**Example**
translations___dutch___yoast_seo__for_wordpress____glotpress
The entered translation is `Het focus-zoekwoord '%1$s' komt niet voor in de pagina titel.`. It's added as expected. However, when going back to the string, or when looking at the waiting strings, it shows as
`Het focus-zoekwoord \'%1$s\' komt niet voor in de pagina titel.`.
(link to [live example](http://translate.yoast.com/gp/projects/wordpress-seo/nl/default/?filters[status]=either&filters[original_id]=4123&sort[by]=translation_date_added&sort[how]=asc))

That's annoying, but in [this string](http://translate.yoast.com/gp/projects/wordpress-seo-premium/nl/default/?filters%5Bterm%5D=&filters%5Buser_login%5D=&filters%5Bstatus%5D=waiting&filter=Filter&sort%5Bby%5D=priority&sort%5Bhow%5D=asc), things are getting problematic:
`Het redirect-type is de HTTP respons code die naar de browser wordt gezonden om aan te geven welk type redirect er wordt uitgeserveerd. <br/><br/>Lees <a href=\'%s\' target=\'_blank\'>deze pagina</a> voor meer informatie.` means the HTML is affected and thus causes invalid HTML output.

**Possible fix**
Do not escape every input.

**Tested versions**
WordPress 4.4.1
GlotPress plugin 1.0

### PR DESCRIPTION
**Pull request by toolstack**
Resolves #220.

### CODE REVIEW

File: gp-includes/cli.php [hunk 1]
```diff
@@ -41,7 +41,7 @@ function __construct() {
 		_deprecated_function( 'GP_CLI::__construct', '', 'WP_CLI_Command' );
 
 		global $argv;
-		if ( gp_array_get( $_SERVER, 'HTTP_HOST' ) ) {
+		if ( wp_unslash( gp_array_get( $_SERVER, 'HTTP_HOST' ) ) ) {
```
Comment (ocean90) [inline gp-includes/cli.php]: `GP_CLI` is deprecated, to keep the change clean let's ignore this file. It's also irrelevant whether the value has slashes or not.

File: gp-includes/router.php [hunk 2]
```diff
@@ -28,7 +28,7 @@ public function request_uri() {
 	}
 
 	public function request_method() {
-		return gp_array_get( $_SERVER, 'REQUEST_METHOD', 'GET' );
+		return wp_unslash( gp_array_get( $_SERVER, 'REQUEST_METHOD', 'GET' ) );
```
Comment (ocean90) [inline gp-includes/router.php:31]: I don't think that `$_SERVER['REQUEST_METHOD']` will ever include quotes/slashes.

File: gp-includes/cli.php (hunk 1 above)
Comment (toolstack) [inline gp-includes/cli.php]: Just added it to all the locations for completeness.  I'll remove it from the PR.

File: gp-includes/router.php (hunk 2 above)
Comment (toolstack) [inline gp-includes/router.php:31]: Neither do I but as WordPress adds slashes to it, it's best to strip them out just in case.

Comment (ocean90) [inline gp-includes/router.php:31]: Well, they are only added for something like `GET"FOO`, but GP_Router->route() neither supports `GET"FOO` or `GET\"FOR`.

Comment (toolstack) [inline gp-includes/router.php:31]: I'll remove it if you like, but as it doesn't add much overhead and future proofs us it seems like a reasonable thing to do.

**Comment by ocean90** [comment]
Merged to hotfix-1.0.1 in https://github.com/GlotPress/GlotPress-WP/commit/8e9d6524401098865b32dc5d6dfe602e55d41792.

**Pull request merged**